shared/nft-collection.ts

```diff
@@ -201,8 +202,7 @@ export class NFTCollectionHelper extends DeployHelpers {
   }
 }
 
-async function calcContractId(txId: string, groupIndex: number) {
-  const nodeProvider = web3.getCurrentNodeProvider()
+async function calcContractId(nodeProvider: NodeProvider, txId: string, groupIndex: number) {
```

Comment:
Seems that this doesn't work well on non-devnet environment, since sometimes it takes a while before we can query tx details.

Reply:
Great catch, thanks! Now we get the number of fixed outputs from `decode-unsigned-tx` endpoint, we need to expose this endpoint on testnet/mainnet.